xformers/factory/block_factory.py

```diff
@@ -303,6 +302,13 @@ def __init__(self, config: xFormerEncoderConfig, **kwargs):
             else None
         )
 
+        if self.pose_encoding is not None:
+            pos_encoding_dim = config.position_encoding_config.dim_model
+            mha_dim = config.multi_head_config["dim_model"]
+
+            if pos_encoding_dim != mha_dim:
+                self.embedding_projector = nn.Linear(pos_encoding_dim, mha_dim)
```

Comment:
could you add a warning here ? (logging.warning) It will be only visible at construction time, so no spamming, but could make it easier to debug if this is not on purpose